### PR TITLE
Fix __call__ Overload Types

### PR DESCRIPTION
**Pull request by needuv**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_service_groundedness/_service_groundedness.py [hunk 1]
```diff
@@ -105,18 +105,18 @@ def __init__(
     def __call__(
         self,
         *,
+        response: str,
+        context: str,
         query: Optional[str] = None,
```
Comment (minthigpen) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_service_groundedness/_service_groundedness.py:110]:
```suggestion
        query: str,
```

Comment (singankit) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_service_groundedness/_service_groundedness.py:110]:
```suggestion
        query: Optional[str],
```

Comment (minthigpen) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_service_groundedness/_service_groundedness.py:110]: the service based evaluator currently requires query -- we want to make it optional in the future but for now it should be required.

**Review by singankit — APPROVED** [review]

**Comment by needuv** [comment]
Only error was mypy, tests were passing in last iteration. Overriding and merging

**Pull request merged**